src/boimmgpy/model_annotation/model_annotator.py

```diff
@@ -0,0 +1,382 @@
+            lipid_class_dict[backbone] += 1
+            self.backbone = backbone
+            counter,results = self.search_lipid_synonyms(metabolite,side_chain)
+            
+
+        sorted_lipid_class_dict = sorted(
+            lipid_class_dict.items(), key=lambda x: x[1], reverse=True
+        )
+        converted_lipid_class_dict = dict(sorted_lipid_class_dict)
+
+        return (
+            converted_lipid_class_dict,
+            check_annotation_dict,
+            counter,
+            results,
+            
+        )
+
+    def annotation_checker(self, lipid):
+        """Method that checks if a given lipid is annotated in the model
+
+        Args:
+            lipid (_type_): Given lipid metabolite from the model
+        """
+        check_annotation_dict = {}
+        annotation = lipid.annotation.keys()
+        annotated = False
+        check_annotation_dict[lipid.id] = annotated
+        if "slm" in annotation or "lipidmaps" in annotation:
+            annotated = True
+            check_annotation_dict[lipid.id] = annotated
+        return check_annotation_dict
+
+    def search_lipid_synonyms(self, metabolite:Metabolite,side_chain:List[str]):
+        """Method that implements the screening in the database for accurate lipid structure.
+        This method calls all StaticMethods to do the screening in the database.
+
+        Args:
+            metabolite (Metabolite): Given Metabolite of the cobra model
+            side_chain (List[str]): List of the side chains referent to the given Metavolite
+
+        Returns:
+            _type_: tuple with two dictionaries, first one for the extend of the lipid classes that the algorithm caugth and second one for the identifiers to do the annotation
+        """
+        session = self.login()
+        results={}
+        counter = defaultdict(int)
+        is_compound = False
+        lipid_id = self.get_synonym_id(session,self.backbone, side_chain)
+        if lipid_id != None and not lipid_id[2]:
+            for backbone in lipid_id[0]:
+                lipid_compound = self.get_coumpound(session,backbone, lipid_id[1], is_compound)
+                if lipid_compound != None and lipid_compound[0] != 0:
+                    structurally_defined_lipids = (
+                        self.get_compounds_with_specific_parent_set_of_components(
+                        session,lipid_compound[0], lipid_compound[1]
+                        )
+                    )
+                    if structurally_defined_lipids:
+                        if metabolite.id in results:
+                            results[metabolite.id] = (
+                                results[metabolite.id]
+                                + structurally_defined_lipids
+                            )
+                        else:
+                            results[metabolite.id] = structurally_defined_lipids
+                        counter[self.backbone] += 1
+
```

Comment:
The code above seems the same as in the below, only changing few variables... wrap into a private method and call it here to reduce the code duplication